operate/client/src/App/ProcessInstance/BottomPanel/VariablePanel/index.tsx

```diff
@@ -42,27 +42,25 @@ const VariablePanel: React.FC<Props> = observer(function VariablePanel({
   const [listenerTypeFilter, setListenerTypeFilter] =
     useState<ListenerEntity['listenerType']>();
 
-  const shouldUseFlowNodeId = !flowNodeInstanceId && flowNodeId;
-
-  let jobsFilter = {};
-  if (shouldUseFlowNodeId) {
-    jobsFilter = {
-      processInstanceKey: {$eq: processInstanceId},
-      elementId: {$eq: flowNodeId},
-      ...(listenerTypeFilter && {kind: {$eq: listenerTypeFilter}}),
-    };
-  } else if (flowNodeInstanceId) {
-    jobsFilter = {
-      processInstanceKey: {$eq: processInstanceId},
-      elementId: {$eq: flowNodeInstanceId},
-      ...(listenerTypeFilter && {kind: {$eq: listenerTypeFilter}}),
-    };
-  }
-
   // TODO: remove when listeners tab is fully migrated to v2
   const {listenersFailureCount, fetchListeners, reset} =
     processInstanceListenersStore;
 
+  const jobsFilter = useMemo<JobsFilter | undefined>(() => {
+    if (!flowNodeInstanceId && !flowNodeId) {
+      // We cannot meaningfully filter for a flow-node if none is selected.
+      // Therefore, no filter is defined to not trigger a job search.
+      return;
+    }
```

Comment:
I think we're using `jobsFilter` for 2 purposes here and should probably split it into.

I would keep the logic where we set the filter object and add a new variable called `const shouldFetchListeners = !flowNodeInstanceId && !flowNodeId`

This way things are clearly separate and you don't even need the comment to explain this logic